pkg/storage/cache.go

```diff
@@ -17,20 +19,70 @@ const (
 	dbCacheLockCheckTimeout = 10 * time.Second
 )
 
-type Cache struct {
+type Driver struct {
 	rootDir     string
 	db          *bbolt.DB
 	log         zlog.Logger
 	useRelPaths bool // weather or not to use relative paths, should be true for filesystem and false for s3
 }
 
-// Blob is a blob record.
-type Blob struct {
-	Path string
+type BoltDBDriverParameters struct {
+	RootDir, Name string
+	UseRelPaths   bool
 }
 
-func NewCache(rootDir string, name string, useRelPaths bool, log zlog.Logger) *Cache {
-	dbPath := path.Join(rootDir, name+DBExtensionName)
+const (
+	driverName         = "boltdb"
+	defaultRootDir     = "/tmp/zot/boltDB"
+	defaultName        = "cacheBoltDB"
+	defaultUseRelPaths = false
+)
+
+// Implements the storage.DatabaseDriverFactory interface.
+type boltDBDriverFactory struct{}
+
+func (factory *boltDBDriverFactory) Create(parameters interface{}, log zlog.Logger) (cachedb.Driver, error) {
+	properParameters, ok := parameters.(BoltDBDriverParameters)
+	if !ok {
+		panic("Failed type assertion")
+	}
+
+	return FromParameters(properParameters, log)
+}
+
+// nolint:gochecknoinits
```

Comment:
We have managed to do without init() funcs so far. Let's not start now. Call an func inline from initialization path directly.